src/vehicle.cpp

```diff
@@ -1756,7 +1756,6 @@ bool CanBuildVehicleInfrastructure(VehicleType type, byte subtype)
 	assert(IsCompanyBuildableVehicleType(type));
 
 	if (!Company::IsValidID(_local_company)) return false;
-	if (!_settings_client.gui.disable_unsuitable_building) return true;
```

Comment:
I don't understand this change. I think "removing disable_unsuitable_building" should mean "set it to false".
So this line and everything below should be replaced with "return true".